streams/src/test/java/org/apache/kafka/streams/integration/ConsistencyVectorIntegrationTest.java

```diff
@@ -131,67 +129,56 @@ public void shouldHaveSamePositionBoundActiveAndStandBy() throws Exception {
-
-        public File getDbDir() {
-            return dbDir;
-        }
+        final StateQueryRequest<ValueAndTimestamp<Integer>> request =
+            StateQueryRequest
+                .inStore(TABLE_NAME)
+                .withQuery(KeyQuery.<Integer, ValueAndTimestamp<Integer>>withKey(key))
+                .withPositionBound(PositionBound.unbounded());
+
+        checkPosition(batch1NumMessages, request, kafkaStreams1);
+        checkPosition(batch1NumMessages, request, kafkaStreams2);
     }
 
-    public class TestingRocksDbKeyValueBytesStoreSupplier implements KeyValueBytesStoreSupplier {
-
-        public List<TestingRocksDBStore> stores = new LinkedList<>();
-
-        private final String name;
-
-        public TestingRocksDbKeyValueBytesStoreSupplier(final String name) {
-            this.name = name;
-        }
-
-        @Override
-        public String name() {
-            return name;
-        }
+    private void checkPosition(final int batch1NumMessages,
+                               final StateQueryRequest<ValueAndTimestamp<Integer>> request,
+                               final KafkaStreams kafkaStreams1) throws InterruptedException {
+        final long maxWaitMs = TestUtils.DEFAULT_MAX_WAIT_MS;
+        final long expectedEnd = System.currentTimeMillis() + maxWaitMs;
+
+        while (true) {
+            final StateQueryResult<ValueAndTimestamp<Integer>> stateQueryResult =
+                IntegrationTestUtils.iqv2WaitForResult(
+                    kafkaStreams1,
+                    request
+                );
+            final QueryResult<ValueAndTimestamp<Integer>> queryResult =
+                stateQueryResult.getPartitionResults().get(0);
+            if (queryResult.isSuccess() && queryResult.getResult() != null) {
+                // invariant: each value is also at the equivalent offset
+                assertThat(
+                    "Result:" + queryResult,
+                    queryResult.getPosition(),
+                    is(
+                        Position.emptyPosition()
+                                .withComponent(INPUT_TOPIC_NAME, 0, queryResult.getResult().value())
+                    )
+                );
+
+                if (queryResult.getResult().value() == batch1NumMessages - 1) {
+                    // we're at the end of the input.
+                    return;
+                }
+            } else {
+                if (expectedEnd <= System.currentTimeMillis()) {
+                    throw new RuntimeException(
+                        "Test timed out in " + maxWaitMs);
+                }
+            }
 
-        @Override
-        public KeyValueStore<Bytes, byte[]> get() {
-            final TestingRocksDBStore testingRocksDBStore = new TestingRocksDBStore(name, metricsScope());
-            stores.add(testingRocksDBStore);
-            return testingRocksDBStore;
+            // we're not done yet, so sleep a bit and test again.
+            Thread.sleep(maxWaitMs / 10);
```

Comment:
So we basically do 10 re-tries? Is this intended?

Reply:
Yep. This is more attempts than the previous utility, which I think would only have done four retries, but it's the same overall time limit on the test.